BepInEx.Core/Contract/Attributes.cs

```diff
@@ -63,47 +64,49 @@ private static Version TryParseLongVersion(string version)
         return null;
     }
 
-    internal static BepInPlugin FromCecilType(TypeDefinition td)
+    internal static BepInMetadataAttribute FromCecilType(TypeDefinition td)
     {
-        var attr = MetadataHelper.GetCustomAttributes<BepInPlugin>(td, false).FirstOrDefault();
+        var attr = MetadataHelper.GetCustomAttributes<BepInMetadataAttribute>(td, false).FirstOrDefault();
 
         if (attr == null)
             return null;
 
-        return new BepInPlugin((string) attr.ConstructorArguments[0].Value,
-                               (string) attr.ConstructorArguments[1].Value,
-                               (string) attr.ConstructorArguments[2].Value);
+        return new BepInMetadataAttribute((string) attr.ConstructorArguments[0].Value,
+                                          (string) attr.ConstructorArguments[1].Value,
+                                          (string) attr.ConstructorArguments[2].Value);
     }
 }
 
 /// <summary>
-///     This attribute denotes that a class is a plugin provider, and specifies the required metadata.
+///     This attributes defines the phase that a <see cref="Plugin"/>'s <see cref="Plugin.Load"/> method will be called
 /// </summary>
 [AttributeUsage(AttributeTargets.Class)]
-public class BepInPluginProvider : BepInPlugin
+public class BepInPhaseAttribute : Attribute
```

Comment:
Is the phase attribute supposed to be unique per plugin? If yes, it would make sense to combine it with `BepInMetadataAttribute` to ensure people don't forget it.

Reply:
Along with the general comment, added a discussion item. I feel like it shouldn't be part of metadata, but I like the idea to default it to GameInitialised